crates/anvil/src/eth/backend/mem/mod.rs

```diff
@@ -1180,7 +1180,13 @@ impl Backend {
         if let Some(factory) = &self.precompile_factory {
             inject_precompiles(&mut evm, factory.precompiles());
         }
+        let cheats_arc = Arc::new(self.cheats.clone());
 
+        let cheat_ecrecover = CheatEcrecover::new(Arc::clone(&cheats_arc));
+
+        evm.precompiles_mut().apply_precompile(&EC_RECOVER, move |_| {
+            Some(DynPrecompile::new_stateful(move |input| cheat_ecrecover.call(input)))
+        });
```

Comment:
oh one last nit here, we can skip this step entirely if we dont have any cheatcodes, so it doesnt hurt to add an additional check for has_recover_overrides